imap_processing/swe/l1b/swe_l1b.py

```diff
@@ -1,48 +1,923 @@
+    nuclear physics, radiation detection, and particle counting, to compensate
+    for the effects of the time period during which a detector is not able to
+    record new events or measurements after detecting a previous event.
+    This "deadtime" is essentially the time during which the detector is
+    recovering from the previous detection and is unable to detect new events.
+
+    In particle detectors, there is a finite time required for the detector to
+    reset or recover after detecting a particle. During this deadtime, any
+    subsequent particles that may have arrived go undetected. As a result,
+    the recorded count rate appears to be lower than the actual count rate.
+
+    Deadtime correction involves mathematically adjusting the measured count
+    rates to compensate for this deadtime effect. This correction is crucial
+    when dealing with high-intensity sources or particle fluxes, as the deadtime
+    can significantly affect the accuracy of the measurements.
+
+    Deadtime correction is important to ensure accurate measurements and data
+    analysis in fields where event detection rates are high and where every
+    detected event is critical for understanding physical processes.
+
+    Parameters
+    ----------
+    counts : numpy.ndarray
+        Counts data before deadtime corrections.
+    acq_duration : int or numpy.ndarray
+        This is ACQ_DURATION from science packet. acq_duration is in microseconds.
+
+    Returns
+    -------
+    corrected_count : numpy.ndarray
+        Corrected counts.
+    """
+    # deadtime is 360 ns
+    deadtime = 360e-9
+    if isinstance(acq_duration, int):
+        # Convert acq_duration to a numpy array for consistency
+        acq_duration = np.array([acq_duration])
+    correct = 1.0 - (deadtime * (counts / (acq_duration[..., np.newaxis] * 1e-6)))
+    correct = np.maximum(0.1, correct)
+    corrected_count = counts.astype(np.float64) / correct
+    return corrected_count
+
+
+def convert_counts_to_rate(data: np.ndarray, acq_duration: np.ndarray) -> npt.NDArray:
+    """
+    Convert counts to rate using sampling time.
+
+    acq_duration is ACQ_DURATION from science packet.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Counts data.
+    acq_duration : numpy.ndarray
+        Acquisition duration. acq_duration is in microseconds.
+
+    Returns
+    -------
+    numpy.ndarray
+        Count rates array in seconds.
+    """
+    # Convert microseconds to seconds without modifying the original acq_duration
+    acq_duration_sec = acq_duration * 1e-6
+
+    # Ensure acq_duration_sec is broadcastable to data
+    if acq_duration_sec.ndim < data.ndim:
+        acq_duration_sec = acq_duration_sec[
+            ..., np.newaxis
+        ]  # Add a new axis for broadcasting
+
+    # Perform element-wise division
+    count_rate = data / acq_duration_sec
+    return count_rate.astype(np.float64)
```

Comment:
To preserve precision, you should upcast before the division.

```suggestion
    count_rate = data.astype(np.float64) / acq_duration_sec
    return count_rate
```